nectar/src/database.rs

```diff
@@ -142,14 +142,15 @@ impl Database {
                     let swap = deserialize::<Swap>(&value).context("failed to deserialize swap");
 
                     match (swap_id, swap) {
-                        (Ok(swap_id), Ok(swap)) => Some(Ok(SwapKind::from((swap, swap_id)))),
+                        (Ok(swap_id), Ok(swap)) => Some(Ok((swap, swap_id))),
                         (Ok(_), Err(err)) => Some(Err(err)), // If the swap id deserialize, then
                         // it should be a swap
                         (..) => None, // This is not a swap item
                     }
                 }
                 Err(err) => Some(Err(err).context("failed to retrieve swaps from DB")),
             })
+            .map(|res| res.map(|(swap, swap_id)| SwapKind::from((swap, swap_id))))
```

Comment:
Does clippy not suggest you to use a function reference here?

```suggestion
            .map(|res| res.map(SwapKind::from))
```

Reply:
You can see the CI passed :)

Reply:
Will fix with follow up PR